src/me/mrCookieSlime/Slimefun/listeners/AncientAltarListener.java

```diff
@@ -171,5 +171,32 @@ public void onBlockPlace(BlockPlaceEvent e) {
 			e.setCancelled(true);
 		}
 	}
-}
-
+	
+	@EventHandler (priority = EventPriority.HIGHEST, ignoreCancelled = true)
+	public void onBlockBreak(BlockBreakEvent e) {
+		Block b = e.getBlock();
+		SlimefunItem item = BlockStorage.check(b);
+		if(item == null) return;
+		if(item.getID().equalsIgnoreCase("ANCIENT_PEDESTAL")) {
+			Item stack = findItem(b);
+			if(stack == null) return;
+			b.getWorld().dropItem(b.getLocation(), fixItemStack(stack.getItemStack(), stack.getCustomName()));
+			stack.remove();
+		}
+	}
+	
+	@EventHandler (priority = EventPriority.HIGHEST, ignoreCancelled = true)
+	public void onBlockExplode(EntityExplodeEvent e) {
+		for(Block b:e.blockList()) {
+			SlimefunItem item = BlockStorage.check(b);
+			if(item == null) continue;
+			if(item.getID().equalsIgnoreCase("ANCIENT_PEDESTAL")) {
+				Item stack = findItem(b);
+				if(stack == null) return;
+				if(SlimefunStartup.getCfg().getBoolean("options.drop-item-when-ancient-pedestal-explodes"))
+					b.getWorld().dropItem(b.getLocation(), fixItemStack(stack.getItemStack(), stack.getCustomName()));
+				stack.remove();
+			}
+		}
+	}
+}
```

Comment:
for all of the above:

**https://github.com/TheBusyBiscuit/Slimefun4/blob/6f61176d0c9cb944976e95813160a8df0759effc/src/me/mrCookieSlime/Slimefun/Objects/SlimefunItem/handlers/BlockBreakHandler.java**

Reply:
Nevermind, it's this class:

https://github.com/TheBusyBiscuit/Slimefun4/blob/6f61176d0c9cb944976e95813160a8df0759effc/src/me/mrCookieSlime/Slimefun/Objects/SlimefunBlockHandler.java

Reply:
Could also be this class:

https://github.com/TheBusyBiscuit/Slimefun4/blob/6f61176d0c9cb944976e95813160a8df0759effc/src/me/mrCookieSlime/Slimefun/Objects/SlimefunBlockHandler.java